FirebaseInstallations/Source/Tests/FIRInstallationsTests.m

```diff
@@ -0,0 +1,132 @@
+        XCTAssertNil(error);
+        XCTAssertNotNil(identifier);
+        XCTAssertGreaterThan(identifier.length, 0);
+
+        [idExpectation fulfill];
+      }];
+
+  [self waitForExpectations:@[ idExpectation ] timeout:0.5];
+}
+
+- (void)testAuthTokenSuccess {
+  FIRInstallations *installations = [self assertInstallationsWithAppNamed:@"app"];
+
+  XCTestExpectation *tokenExpectation = [self expectationWithDescription:@"AuthTokenSuccess"];
+  [installations authTokenWithCompletion:^(FIRAuthTokenResult *_Nullable tokenResult,
+                                           NSError *_Nullable error) {
+    XCTAssertNotNil(tokenResult);
+    XCTAssertGreaterThan(tokenResult.authToken.length, 0);
+    XCTAssertTrue([tokenResult.expirationTime laterDate:[NSDate date]]);
+    XCTAssertNil(error);
+
+    [tokenExpectation fulfill];
+  }];
+
+  [self waitForExpectations:@[ tokenExpectation ] timeout:0.5];
+}
+
+- (void)testAuthTokenForcingRefreshSuccess {
+  FIRInstallations *installations = [self assertInstallationsWithAppNamed:@"app"];
+
+  XCTestExpectation *tokenExpectation = [self expectationWithDescription:@"AuthTokenSuccess"];
+  [installations authTokenForcingRefresh:YES
+                              completion:^(FIRAuthTokenResult *_Nullable tokenResult,
+                                           NSError *_Nullable error) {
+                                XCTAssertNotNil(tokenResult);
+                                XCTAssertGreaterThan(tokenResult.authToken.length, 0);
+                                XCTAssertTrue([tokenResult.expirationTime laterDate:[NSDate date]]);
+                                XCTAssertNil(error);
+
+                                [tokenExpectation fulfill];
+                              }];
+
+  [self waitForExpectations:@[ tokenExpectation ] timeout:0.5];
+}
+
+- (void)testDeleteSuccess {
+  FIRInstallations *installations = [self assertInstallationsWithAppNamed:@"app"];
+
+  XCTestExpectation *deleteExpectation = [self expectationWithDescription:@"DeleteSuccess"];
+  [installations deleteWithCompletion:^(NSError *_Nullable error) {
+    XCTAssertNil(error);
+    [deleteExpectation fulfill];
+  }];
+
+  [self waitForExpectations:@[ deleteExpectation ] timeout:0.5];
+}
+
+#pragma mark - Common
+
+- (FIRInstallations *)assertInstallationsWithAppNamed:(NSString *)appName {
+  FIRApp *app = [self createAndConfigureAppWithName:appName];
+  FIRInstallations *installations = [FIRInstallations installationsWithApp:app];
+
+  XCTAssertEqualObjects(installations.appID, app.options.googleAppID);
+  XCTAssertEqualObjects(installations.appName, app.name);
+
+  return installations;
+}
+
+#pragma mark - Helpers
+
+- (FIRApp *)createAndConfigureAppWithName:(NSString *)name {
+  FIROptions *options = [[FIROptions alloc] initInternalWithOptionsDictionary:@{
+    @"GOOGLE_APP_ID" : @"1:1085102361755:ios:f790a919483d5bdf",
+  }];
+  [FIRApp configureWithName:name options:options];
```

Comment:
`configureWithName:options:` has some internal state associated with it that may affect tests running concurrently that can lead to flakes. Using the `[FIRInstallations installations]` singleton can do the same. I mentioned a way to get around this in the other file about the initializer, then no FIRApp is necessary for most

Reply:
Good point. I have couple questions here:
1. Will I need to configure a `FIRApp` to test singleton? 
2. Do we run tests concurrently?

Reply:
1. Yes, that is required. There are ways to do it without explicitly calling `configure` but if it's a single test case that's probably okay and reduces the surface for flakes due to global state. I believe we have some other code to have the container properly initialized without global state but I'll have to find it if needed.
2. Yes. Tests in other files run concurrently, but test cases in the same class run synchronously.

Reply:
I kept the logic for the singleton tests but other tests use `initWithGoogleAppID:appName:`